src/Illuminate/Testing/Testing.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace Illuminate\Testing;
+
+use Illuminate\Contracts\Foundation\Application;
+
+class Testing
+{
+    /**
+     * The application instance.
+     *
+     * @var \Illuminate\Contracts\Foundation\Application
+     */
+    protected $app;
+
+    /**
+     * The token resolver callback.
+     *
+     * @var \Closure|null
+     */
+    protected static $tokenResolver;
+
+    /**
+     * Create a new Testing instance.
+     *
+     * @param  \Illuminate\Contracts\Foundation\Application $app
+     * @return void
+     */
+    public function __construct(Application $app)
+    {
+        $this->app = $app;
+    }
+
+    /**
+     * Adds an unique test token to the given string, if needed.
+     *
+     * @return string
+     */
+    public function addTokenIfNeeded($string)
+    {
+        if (! $this->inParallel()) {
+            return $string;
+        }
+
+        return "{$string}_test_{$this->token()}";
+    }
+
+    /**
+     * Apply the callback if tests are running in parallel.
+     *
+     * @param  callable $callback
+     * @return void
+     */
+    public function whenRunningInParallel($callback)
+    {
+        if ($this->inParallel()) {
+            $callback();
+        }
+    }
+
+    /**
+     * Indicates if the current tests are been run in Parallel.
+     *
+     * @return bool
+     */
+    protected function inParallel()
+    {
+        return $this->app->runningUnitTests() && ! empty($_SERVER['LARAVEL_PARALLEL_TESTING']) && $this->token();
```

Comment:
Could the restult of `token()` ever be 0? because that would also be falsy here.

Reply:
No. It's always a positive number.

Reply:
I think @BackEndTea  has an interesting concern given we have a userland callback been run [here](https://github.com/laravel/framework/pull/35778/files#diff-8210adc9a77de4eebc03bd70c12fc108acd07942e208efebfce1b254beab6526R98)

Reply:
@gocanto Just removed that method from the Public API: https://github.com/laravel/framework/pull/35778/commits/2f60c87d976653ab27ff5e1cefdfa0fdfba7d579. That method shouldn't really be run in userland..